core/src/marker/markerManager.cpp

```diff
@@ -14,15 +15,20 @@
 
 namespace Tangram {
 
+MarkerManager::MarkerManager() {}
+
+MarkerManager::~MarkerManager() {}
+
 void MarkerManager::setScene(std::shared_ptr<Scene> scene) {
 
     m_scene = scene;
     m_mapProjection = scene->mapProjection().get();
-    m_styleContext.initFunctions(*scene);
-    m_jsFnIndex = scene->functions().size();
 
+    m_styleContext = std::make_unique<StyleContext>();
+    m_styleContext->initFunctions(*scene);
 
     // Initialize StyleBuilders.
+    m_styleBuilders.clear();
```

Comment:
I'm not sure if it's really necessary to remove all markers on scene load. 
It's the question whether the api for loadScene is more friendly when it states that after scene loading all markers are invalidated and need to be recreated or that markers which use custom styles not defined in the new scene need to be updated with a new styling.

I think it's more common to use markers that use the default styles and it would be nice if we could keep that case simple.